test/inherited_provider_test.dart

```diff
@@ -2408,15 +2400,47 @@ DeferredInheritedProvider<int, int>(controller: 42, value: 24)'''),
     expect(find.text('true true'), findsOneWidget);
     expect(buildCount, 4);
   });
+
+  testWidgets('throws ProviderNotFoundException if lookup fails within create',
```

Comment:
We need a test for what happens when we call `context.watch` after a provider threw